src/controllers/tag.js

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable require-jsdoc */
+// import isEmpty from 'utils';
+import models from '../models';
+import Util from '../helpers/util';
+import result from '../helpers/tags';
+
+const util = new Util();
+const internalError = () => {
+  util.setError(500, 'Internal Error');
+  return util;
+};
+
+const notFound = (msg) => {
+  util.setError(404, `${msg} not found`);
+  return util;
+};
+
+const addTags = (additions) => {
+  result.addArticleTags(additions);
+  return result;
+};
+
+const {
+  Tag, Article, ArticleTag
+} = models;
+
+class TagController {
+  static async createTag(req, res) {
+    // create orphan tag
+    const { name } = req.body;
+    Tag.findOrCreate({
+      where: { name }
+    }).then((tag) => {
+      if (tag[1]) {
+        return res.status(201).json({ message: `tag ${name} created`, data: tag[0] });
+      }
+      res.status(200).json({ message: `tag ${name} exists`, data: tag[0] });
+    }).catch(() => {
+      internalError.send(res);
+    });
+  }
+
+  static async createArticleTag(req, res) {
+    const tags = req.body.name.split(', ');
+    const article = await Article.findOne({
+      where: { id: req.params.articleId },
+    });
+    const currentTags = await article.getTags();
+    const added = tags.filter(elt => !currentTags.map(x => x.name).includes(elt));
+    const exists = tags.filter(elt => currentTags.map(x => x.name).includes(elt));
+    tags.forEach((name) => {
+      Tag.findOrCreate({
+        where: { name }
+      }).then(tag => article.addTag(tag[0]));
+    });
+    return addTags([added, exists, article]).send(res);
+  }
+
+  static async editTag(req, res) {
+    // changes the name of a given tag if new name still unique or merges tags
```

Comment:
You could use doc-strings for a multiple line comment

Reply:
Thanks for the observations @nkalyesubula, implemented